blog/views.py

```diff
@@ -0,0 +1,60 @@
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.http import HttpRequest, HttpResponse
+from django.shortcuts import render, redirect
+from django.urls import reverse_lazy
+from django.views import generic
+
+from blog.forms import CommentaryForm
+from blog.models import Post, Commentary
+
+
+class IndexListView(generic.ListView):
+    model = Post
+    queryset = Post.objects.select_related("owner")
```

Comment:
models and querysets are interchangeable, so you should provide either model or queryset, not both.